src/store/PeopleContext.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useState, useMemo, useCallback } from 'react';
+import { Person } from '../types';
+import { getPeople } from '../api';
+
+export const PeopleContext = React.createContext({
+  people: [] as Person[],
+  loading: false,
+  errorMessage: '',
+  loadPeople: async () => {},
+});
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const PeopleProvider: React.FC<Props> = ({ children }) => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const loadPeople = useCallback(() => {
+    setLoading(true);
+
+    return getPeople()
+      .then(setPeople)
+      .catch(() => setErrorMessage('Something went wrong'))
```

Comment:
The error message is hardcoded as 'Something went wrong'. Consider using the error message from the catch block if available, to provide more specific feedback to the user.